crates/iota-swarm-config/src/node_config_builder.rs

```diff
@@ -328,8 +330,8 @@ impl FullnodeConfigBuilder {
         self
     }
 
-    pub fn with_admin_interface_port(mut self, admin_interface_port: u16) -> Self {
-        self.admin_interface_port = Some(admin_interface_port);
+    pub fn with_admin_interface_address(mut self, admin_interface_address: SocketAddr) -> Self {
+        self.admin_interface_address = Some(admin_interface_address);
         self
     }
```

Comment:
```suggestion
    pub fn with_admin_interface_address(mut self, admin_interface_address: impl Into<SocketAddr>) -> Self {
        self.admin_interface_address = Some(admin_interface_address.into());
        self
    }
```

You can do this here and in the `run_admin_server`. There are a lot of useful `Into<SocketAddr>` impls so this is pretty convenient. Ex:

```rust
.with_admin_interface_address("127.0.0.1:1337".parse().unwrap())
```

becomes

```rust
.with_admin_interface_address(([127, 0, 0, 1], 1337))
```